ext/mysqlnd/mysqlnd_connection.c

```diff
@@ -543,7 +543,11 @@ MYSQLND_METHOD(mysqlnd_conn_data, get_scheme)(MYSQLND_CONN_DATA * conn, MYSQLND_
 		if (!port) {
 			port = 3306;
 		}
-		transport.l = mnd_sprintf(&transport.s, 0, "tcp://%s:%u", hostname.s, port);
+		if (strchr(hostname.s, ':') == NULL) {
```

Comment:
I don't believe this works for ipv6 addresses, because they already contain `:`.
e.g.: `mysqli_connect("[::1]"," root", "", port: 123);` this will not use port 123, it results in `transport.s` being `tcp://[::1]`. Prior to this PR it did use port 123.
But the general idea of the PR is good, it just needs fixing for ipv6 I think.

Reply:
Now that I think again about this, I notice that there's also another existing issue with ipv6 here.
Omitting the square brackets, i.e. `::1` for example should also work as a hostname. But if you specify a port, then `transport.s` becomes `::1:portnr`. This is an invalid ipv6 address.

Reply:
I wonder if it works with IPv6 at all. I found this bug https://bugs.php.net/bug.php?id=67563

Reply:
Looking at the 2nd comment, they seem to suggest it works with brackets.